pip-zipapp.yaml

```diff
@@ -0,0 +1,65 @@
+package:
+  name: pip-zipapp
+  version: 24.0
+  epoch: 0
+  description: "Pip as a python zipapp"
+  copyright:
+    - license: MIT
+
+environment:
+  contents:
+    packages:
+      - busybox
+
+pipeline:
```

Comment:
Could we add some comments for where the various sources below are fetched from, and why. Not being a python expert, it's a bit tricky to reason about it. Things like if there's an update to pypa/pip, how do these get updated.

Reply:
Done, I'll add an issue once merged to see if we can make this clearer